lib/array.rb

```diff
@@ -0,0 +1,16 @@
+class Array
+
+  def molliesinject(start_point = nil)
+    start_point = start_point
+    self.each_with_index do |element, index|
+      if index == 0 && start_point == nil
+        @total = element
+      elsif @start_point != nil
+        self.each { |num| @total = yield(start_point, num) }
```

Comment:
Not sure what's going on here... seems like you have an iteration nested inside an iteration.